File: task3/main.kt
fun main() {
    print("Введите число: ")
    val n = readln().toInt()
    var count = 0

    repeat(n) {
        val num = readln().toInt()
        var temp = num
        val lastDigit = temp % 10
        var same = true
        
        while (temp > 0) {
            if (temp % 10 != lastDigit) {
                same = false
                break
            }
            temp /= 10
        }
        
        if (same) count++
    }

    println("Количество: $count")
}

File: task3/main.php
<?php
// No external dependencies needed

echo "Введите количество чисел: "; //обработка ввода
$n = (int) trim(fgets(STDIN));
$count = 0;

for ($i = 0; $i < $n; $i++) {
    $num = (int) trim(fgets(STDIN));
    $last_digit = $num % 10;
    $num = intdiv($num, 10);
    $same_digits = true;

    while ($num > 0) {
        $current_digit = $num % 10;
        if ($current_digit != $last_digit) {
            $same_digits = false;
            break;
        }
        $num = intdiv($num, 10);
    }

    if ($same_digits) {
        $count++;
    }
}

echo "Количество чисел с одинаковыми цифрами: " . $count . "\n";
?>

